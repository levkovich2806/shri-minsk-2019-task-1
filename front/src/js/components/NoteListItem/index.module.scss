.container {
  position: relative;
  padding-left: 28px;

  margin-bottom: 12px;
  cursor: pointer;
  line-height: 22px;
  font-size: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.checkmark {
  position: absolute;
  top: 0;
  left: 0;
  height: 18px;
  width: 18px;
  background-color: #d9d9d9;
  border-radius: 3px;
}

.container:hover input ~ .checkmark {
  background-color: #ccc;
}

.container input:checked ~ .checkmark {
  background-color: #fc0;
}

.checkmark::after {
  content: "";
  position: absolute;
  display: none;
}

.container .checkmark::after {
  left: 5px;
  top: 1px;
  width: 5px;
  height: 10px;
  border: solid black;
  border-width: 0 3px 3px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.container input:checked ~ .checkmark::after {
  display: block;
}

.container input:checked ~ span {
  text-decoration: line-through;
}
