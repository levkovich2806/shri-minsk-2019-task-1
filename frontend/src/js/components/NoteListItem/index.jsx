import React, { Component } from "react";
import styles from "./index.module.scss";

class NoteListItem extends Component {
  render() {
    const { text, checked } = this.props;
    return (
      <label className={styles.container}>
        <input type="checkbox" defaultChecked={checked} />{" "}
        {text && <span>{text}</span>}
        <span className={styles.checkmark} />
      </label>
    );
  }
}

export default NoteListItem;
