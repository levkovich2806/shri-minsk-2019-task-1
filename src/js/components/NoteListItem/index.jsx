import React, { Component } from "react";
import styles from "./index.module.scss";

class NoteListItem extends Component {
  render() {
    const { text, checked } = this.props;
    const checkedState = checked ? true : false;
    return (
      <label className={styles.container}>
        <input type="checkbox" defaultChecked={checkedState} />{" "}
        <span>{text}</span>
        <span className={styles.checkmark} />
      </label>
    );
  }
}

export default NoteListItem;
