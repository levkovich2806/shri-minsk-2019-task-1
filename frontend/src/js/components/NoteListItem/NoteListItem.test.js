import React from "react";
import NoteListItem from "./index.jsx";
import renderer from "react-test-renderer";

describe("NoteListItem", function () {
  test("NoteListItem rendering", () => {

    const component = renderer.create(<NoteListItem text={'Тестовый текст'} checked={true} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
