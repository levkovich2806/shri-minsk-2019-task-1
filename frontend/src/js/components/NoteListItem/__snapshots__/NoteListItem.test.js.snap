// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteListItem NoteListItem rendering 1`] = `
<label
  className="container"
>
  <input
    defaultChecked={true}
    type="checkbox"
  />
   
  <span>
    Тестовый текст
  </span>
  <span
    className="checkmark"
  />
</label>
`;
